composeApp/src/commonMain/kotlin/org/ooni/probe/domain/GetTestDescriptors.kt

```diff
@@ -2,31 +2,34 @@ package org.ooni.probe.domain
 
 import androidx.compose.runtime.Composable
 import kotlinx.coroutines.flow.Flow
-import kotlinx.coroutines.flow.asFlow
-import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.map
 import ooniprobe.composeapp.generated.resources.Res
 import ooniprobe.composeapp.generated.resources.Settings_TestOptions_LongRunningTest
 import org.jetbrains.compose.resources.stringResource
 import org.ooni.probe.data.models.DefaultTestDescriptor
 import org.ooni.probe.data.models.Descriptor
+import org.ooni.probe.data.models.DescriptorUpdatesStatus
 import org.ooni.probe.data.models.InstalledTestDescriptorModel
+import org.ooni.probe.data.models.UpdateStatus
 import org.ooni.probe.data.models.toDescriptor
 
 class GetTestDescriptors(
     private val getDefaultTestDescriptors: () -> List<DefaultTestDescriptor>,
     private val listInstalledTestDescriptors: () -> Flow<List<InstalledTestDescriptorModel>>,
+    private val descriptorUpdates: () -> Flow<DescriptorUpdatesStatus>,
 ) {
     operator fun invoke(): Flow<List<Descriptor>> {
-        return suspend {
-            getDefaultTestDescriptors()
-                .map { it.toDescriptor() }
-        }.asFlow()
-            .flatMapLatest { defaultDescriptors ->
-                listInstalledTestDescriptors()
-                    .map { list -> list.map { it.toDescriptor() } }
-                    .map { defaultDescriptors + it }
+        return combine(
+            listInstalledTestDescriptors(),
+            descriptorUpdates(),
+        ) { installedDescriptors, descriptorUpdates ->
+            val updatedDescriptors = installedDescriptors.map { item ->
+                item.toDescriptor(updateStatus = descriptorUpdates.getStatusOf(item.id))
             }
+            return@combine getDefaultTestDescriptors()
```

Comment:
This will still fetch the default test descriptors on every DB or Pull down to refresh. Not a big deal because they are in memory, but unnecessary if this call was done previously.